### DISCUSSION TITLE
Any idea what changed in the Android Bitwraden Client beta version?

### DISCUSSION THREAD

**Discussion by Lockszmith-GH** [dani-garcia/vaultwarden#5131] [Q&A]
Web vault, works.
Old android client works.
Windows client works.
Keyguard android client works.

New Android beta client (2024.9.0 | 19209) logs in successfully, sync runs for a short time, but vault is empty.

I've reviewed other thread about issues with the beta client, but nothing points me towards a solution.

Can someone explain what changed in the way the android app does it's communications?
Maybe that will point me in the right direction.

UPDATE: Seems to be _something_ IN the vault.

<details> <summary>Admin Diagnostics</summary>

Diagnostics
Versions

Server Installed Ok
    1.32.2 
Server Latest
    1.32.2 
Web Installed
    2024.6.2c 
Database
    PostgreSQL: PostgreSQL 17.0 (Debian 17.0-1.pgdg120+1) on x86_64-pc-linux-gnu, compiled by gcc (Debian 12.2.0-14) 12.2.0, 64-bit 

Checks

OS/Arch
    linux / x86_64 
Running within a container
    Yes (Base: Debian) 
Environment settings overridden
    No 
Uses a reverse proxy
    Yes 
IP header Match
    Config/Server: X-Real-IP 
Internet access Ok
    Yes 
Internet access via a proxy
    No 
DNS (github.com) Ok
    ###.###.###.###
Date & Time (Local)
    Server: 2024-10-25 20:17:58 +00:00 
Date & Time (UTC) Server/Browser Ok Server NTP Ok Browser NTP Ok
    NTP: 2024-10-25 20:17:58 UTC Server: 2024-10-25 20:17:58 UTC Browser: 2024-10-25 20:18:00 UTC 
Domain configuration Match HTTPS

 </details>

**Comment by BlackDex** [comment]
There must be something in your web-vault which breaks. 
Be sure to use the `latest` version of Vaultwarden. Or try the `testing` tagged version first which has some more fixes.

If even the testing tagged version doesn't work then we would need more details to try to see what is going on. 

Instructions on how to provide these details can be found here https://github.com/dani-garcia/vaultwarden/issues/4870#issuecomment-2306749481

**Reply by Lockszmith-GH** [reply]
Thanks. I cam back to update it's probably about the content of the vault. I'll download the -qa version and see if I can figure something out.

**Reply by Lockszmith-GH** [reply]
Frustrating. The entire communication process completes without issues. It's most likely something with the loading of the entries, but there is no exception caught...

Looks like it's time to spin up Visual Studio and attempt to debug this on the PC.

Very frustrating, this vault is associated with a number of organizations, it's going to be hell to pin-point the issue.

Anyway, thanks for the pointer (BTW, I used a tool called [lognote](https://github.com/cdcsgit/lognote) - it's a wireshark like tool for `adb logcat` with filtering)

**Reply by BlackDex** [reply]
If you can shift through the sync data, and see if types are valid. Like numbers which are within quotes maybe. `null` values on dates maybe.

If you for example have a different account which works, try to compare the sync output.

If you suspect an org, you could try to remove the broken account from each collection one at a time to narrow down where it breaks maybe. And maybe even revoke the user from the organisation and see if it's somewhere in the personal vault.

But again, be sure to use the `testing` tagged container which has several fixes not yet in the final released version.

**Reply by BlackDex** [reply]
And if course if you find it, please share the details. If not, maybe you are willing to share the sync data with me personally so that i can maybe figure it out?

**Reply by Lockszmith-GH** [reply]
I did post a fix a few years back to the mobile app, which used to crash when it encountered bad data.

I'm guessing it's something like that.

Furthermore, I wish there was a linter for the data, something that would test for validity of everything on the backend - would make it easier to pinpoint issues like this.

**Reply by BlackDex** [reply]
Well, we first need to know what is bad before you can make a linter or convert the wrong data to valid data.

Vaultwarden currently tries to convert everything so the clients will work and not crash. But there could of course still be something missed, or some exotic item missed.

**Reply by Lockszmith-GH** [reply]
Fair enough, I'll try and sift through the data, but there are thousands of entries...
I've been using it for a very long time.
I'll start by narrowing which of the organizations is effected, as there are multiple users who are experiencing this.

We are a family of 3 people, I've got a couple of organizations between us, I've got 3 more friends connected each to their own organization. (yes, yes I know, overkill, it's just me experimenting across the years with what's possible with the divisions and permission layers - I'm just that type of nerd).

I know my immediate family is affected, haven't heard from my friends yet, so maybe only us.

I'll report after the weekend. Thanks for responding so promptly
